src/main/java/com/github/madhavdhatrak/blaze4j/BlazeWrapper.java

```diff
@@ -250,6 +251,16 @@ private static MemorySegment customResolver(MemorySegment uriPtrSegment) {
                     return MemorySegment.NULL;
                 }
                 
+                // Handle pre-registered schemas when URI is neither HTTP(S) nor classpath
+                if (uri != null &&
+                    !uri.startsWith("http://") && !uri.startsWith("https://") &&
+                    !uri.startsWith("classpath://") &&
```

Comment:
I think it would be important top remove the startsWith checks. Ie. users should be able to pre-register schemas with http:// URI. This pre-registration feature is sometimes used to pre-fetch schemas, so that they aren't downloaded during schema compilation. But the reference will still remain a http:// URL in these cases.